### PR TITLE
multiple interactions on interactable target

### PR DESCRIPTION
**Pull request by TheEmidee**
Reworked interactions to allow an interactable target to propose multiple options
Support for tag filtering on the target or the instigator
Support for specific input mapping context or input action

### CODE REVIEW

File: Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp [hunk 1]
```diff
@@ -38,6 +43,98 @@ void UGBFGameplayAbility_Interact::UpdateInteractions( const FGameplayAbilityTar
...
+    {
+        if ( context.IsValid() )
+        {
+            context.EnhancedSystem->RemoveMappingContext( context.InputMappingContext.Get() );
+        }
+    }
+
+    InputConfigInfos.Reset();
+
+    for ( const auto context : BindActionHandles )
```
Comment (Frickerson) [inline Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp:66]: also reset array after for loop

File: Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp [hunk 2]
```diff
@@ -38,6 +43,98 @@ void UGBFGameplayAbility_Interact::UpdateInteractions( const FGameplayAbilityTar
...
+
+    for ( const auto context : InputMappingContextInfos )
+    {
+        if ( context.IsValid() )
+        {
+            context.EnhancedSystem->RemoveMappingContext( context.InputMappingContext.Get() );
+        }
+    }
+
+    InputConfigInfos.Reset();
```
Comment (Frickerson) [inline Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp]: InputMappingContextInfos instead

File: Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp [hunk 3]
```diff
@@ -38,6 +43,98 @@ void UGBFGameplayAbility_Interact::UpdateInteractions( const FGameplayAbilityTar
...
+
+    for ( const auto context : BindActionHandles )
+    {
+        TArray< uint32 > handles;
+        handles.Add( context.Handle );
+        context.InputComponent->RemoveBinds( handles );
+    }
+
+    InputConfigInfos.Reset();
```
Comment (Frickerson) [inline Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp]: Already reset earlier

File: Source/GameBaseFramework/Private/Interaction/GBFInteractableActor.cpp [hunk 4]
```diff
@@ -1,12 +1,14 @@
 ﻿#include "Interaction/GBFInteractableActor.h"
 
+#include "OnlineSubsystemTypes.h"
```
Comment (Frickerson) [inline Source/GameBaseFramework/Private/Interaction/GBFInteractableActor.cpp]: <>

File: Source/GameBaseFramework/Private/Interaction/GBFInteractionStatics.cpp [hunk 5]
```diff
@@ -1,5 +1,6 @@
 #include "Interaction/GBFInteractionStatics.h"
 
+#include "AbilitySystemBlueprintLibrary.h"
```
Comment (Frickerson) [inline Source/GameBaseFramework/Private/Interaction/GBFInteractionStatics.cpp]: <>

**Review by Frickerson — CHANGES_REQUESTED** [review]

**Review by ThibNach — DISMISSED** [review]

**Pull request merged**